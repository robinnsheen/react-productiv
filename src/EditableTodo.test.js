import EditableTodo from "./EditableTodo";
import React from "react";
import { render, fireEvent } from "@testing-library/react";


const testTodo = {
  title: "testTitle",
  priority: 2,
  description: "test description"
};

describe("productiv app", function () {
  it("renders without crashing", function () {
    render(<EditableTodo todo={testTodo}/>);
  });

  it("matches snapshot", function () {
    const { container } = render(<EditableTodo todo={testTodo} />);
    expect(container).toMatchSnapshot()
  })

  it("renders with correct info", function () {
    const {container} = render(<EditableTodo todo={testTodo} />);
    expect(container.querySelector(".Todo")).toContainHTML(
      `<div class="Todo">`+
        `<div>`+
          `<b>testTitle</b>`+
         ` <small>(priority: 2)</small>`+
        `</div><div>`+
          `<small>test description</small>`+
        `</div>`+
      `</div>`
    );
  });

  it("renders with correct buttons", function () {
    const {container} = render(<EditableTodo todo={testTodo} />);
    expect(container.querySelector(".Todo")).toContainHTML()
  });

  it("clicking edit button calls update function", function () {
    const mockUpdate = jest.fn();
    const mockRemove = jest.fn();

    let {container, debug} = render(<EditableTodo todo={testTodo} update={mockUpdate} remove={mockRemove} />);

    let editButton = container.querySelector(".EditableTodo-toggle");

    debug(editButton);
    fireEvent.click(editButton);
    //test that we rendered form. consolelog container before and after click
    expect(mockUpdate).toHaveBeenCalledTimes(1);
  });

});
