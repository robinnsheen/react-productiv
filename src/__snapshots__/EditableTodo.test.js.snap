// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`productiv app matches snapshot 1`] = `
<div>
  <div
    class="EditableTodo"
  >
    <div
      class="mb-3"
    >
      <div
        class="float-end text-sm-end"
      >
        <button
          class="EditableTodo-toggle btn-link btn btn-sm"
        >
          Edit
        </button>
        <button
          class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
        >
          Del
        </button>
      </div>
      <div
        class="Todo"
      >
        <div>
          <b>
            testTitle
          </b>

          <small>
            (priority:
            2
            )
          </small>
        </div>
        <div>
          <small>
            test description
          </small>
        </div>
      </div>
    </div>
  </div>
</div>
`;
